instance/tests/js/instance_app_spec.js

```diff
@@ -0,0 +1,121 @@
+// OpenCraft -- tools to aid developing and hosting free software projects
+// Copyright (C) 2015 OpenCraft <xavier@opencraft.com>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+(function(){
+"use strict";
+
+// Tests //////////////////////////////////////////////////////////////////////
+
+describe('Instance app', function () {
+    var $controller,
+        httpBackend,
+        indexController,
+        instanceList,
+        OpenCraftAPI,
+        $scope;
+
+    beforeEach(function() {
+        window.swampdragon = {
+            onChannelMessage: jasmine.createSpy(),
+            ready: jasmine.createSpy()
+        };
+        angular.mock.module('restangular');
+        module('InstanceApp');
```

Comment:
Out of curiosity: What namespace is the second `module` function coming from?  I would have expected that `module` refers to `angular.mock.module`, but I don't know since I'm learning about Angular while doing this review.